source/Irrlicht/CBAWMeshFileLoader.h

```diff
@@ -20,16 +20,18 @@ namespace irr { namespace scene
 class CBAWMeshFileLoader : public IMeshLoader
 {
 private:
-	struct SPair
+	struct SBlobData
 	{
-		core::BlobHeaderV1* header;
-		void* blob;
+		const core::BlobHeaderV1* header;
+		const void* blob;
+		bool wasCreated;
 	};
 
 	struct SContext
```

Comment:
Imagine the BAW file has a huge mesh inside it with auxillary data, or is an archive, it could be couple gigabytes!

We only want to read all the BlobHeaders into RAM, at once and linearly.
Then we can have an std::map<uint64_t,std::pair<BlobHeaderV1*,uint64_t> > which means std::map<handle,std::pair<BlobHeaderV1*,absPosInFile> >

The std::pair could be the SBlobData struct for all I care